scripts/breaking_changes_checker/breaking_changes_tracker.py

```diff
@@ -504,6 +538,52 @@ def check_positional_parameter_added(self, current_parameters_node: Dict) -> Non
                     )
                 )
 
+    def check_non_positional_parameter_added(self, current_parameters_node: Dict) -> None:
+        if current_parameters_node["param_type"] != "positional_or_keyword":
+            if self.class_name:
+                if self.function_name == "__init__":
+                    # This is a new class property defined only through the init and not at the class level
+                    self.features_added.append(
+                        (
+                            self.ADDED_CLASS_PROPERTY_MSG, ChangeType.ADDED_CLASS_PROPERTY,
```

Comment:
Do I understand it right that we assume that if it's a parameter in the init then it's set as an attribute?

Reply:
Yes, although there are exceptions to this case...I'm ok reporting this as a method parameter added instead. Let me know what you think.

Reply:
Thinking about this more....I'll revert here and look back at adding this if it becomes too much of an issue